src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc

```diff
@@ -0,0 +1,238 @@
+                                        mock_policy_handler_,
+                                        mock_allocation_manager_);
+  }
+
+  MessageSharedPtr CreateBasicMessage() {
+    MessageSharedPtr message = CreateMessage();
+    (*message)[application_manager::strings::params]
+              [application_manager::strings::function_id] =
+                  mobile_apis::FunctionID::ButtonPressID;
+    (*message)[application_manager::strings::params]
+              [application_manager::strings::connection_key] = kConnectionKey;
+    (*message)[application_manager::strings::params]
+              [application_manager::strings::connection_key] = kAppId;
+    return message;
+  }
+
+ protected:
+  utils::SharedPtr<MockApplication> mock_app_;
+  MockRequestControlerSettings mock_request_controler;
+  MockProtocolHandler mock_protocol_handler;
+  MockHMIMessageHandler mock_hmi_handler;
+  am::CommandHolderImpl command_holder;
+  testing::NiceMock<rc_rpc_plugin_test::MockResourceAllocationManager>
+      mock_allocation_manager_;
+  smart_objects::SmartObject rc_capabilities_;
+  MockRPCPlugin mock_rpc_plugin;
+  MockCommandFactory mock_command_factory;
+  am::request_controller::RequestController request_controller;
+  am::rpc_service::RPCServiceImpl rpc_service;
+  utils::SharedPtr<RCAppExtension> rc_app_extention_;
+  utils::SharedPtr<am::plugin_manager::MockRPCPluginManager>
+      mock_rpc_plugin_manager;
+};
+
+TEST_F(RCGetInteriorVehicleDataConsentTest,
+       Run_MobileSendButtonPressMessage_HMISendASKDRIVERModeToMobile) {
+  MessageSharedPtr mobile_message = CreateBasicMessage();
+  ON_CALL(mock_policy_handler_, CheckModule(_, _)).WillByDefault(Return(true));
+  EXPECT_CALL(app_mngr_, GetPluginManager())
+      .WillRepeatedly(ReturnRef(*mock_rpc_plugin_manager));
+  utils::Optional<RPCPlugin> rpc_plugin(mock_rpc_plugin);
+  utils::Optional<MockRPCPlugin> mrpc_plugin(mock_rpc_plugin);
+  EXPECT_CALL(*mock_rpc_plugin_manager, FindPluginToProcess(_, _))
+      .WillOnce(Return(rpc_plugin));
+  utils::SharedPtr<RCGetInteriorVehicleDataConsentRequest> rc_consent_request =
+      CreateRCCommand<commands::RCGetInteriorVehicleDataConsentRequest>(
+          mobile_message);
+  EXPECT_CALL(mock_allocation_manager_, IsResourceFree(kResource))
+      .WillOnce(Return(true));
+  EXPECT_CALL(mock_allocation_manager_, AcquireResource(_, _))
+      .WillOnce(Return(rc_rpc_plugin::AcquireResult::ASK_DRIVER));
+  EXPECT_CALL(*mrpc_plugin, GetCommandFactory())
+      .WillOnce(ReturnRef(mock_command_factory));
+  EXPECT_CALL(mock_command_factory, CreateCommand(_, _))
+      .WillOnce(Return(rc_consent_request));
+  EXPECT_CALL(mock_hmi_interfaces_,
+              GetInterfaceFromFunction(
+                  hmi_apis::FunctionID::RC_GetInteriorVehicleDataConsent))
+      .WillOnce(Return(am::HmiInterfaces::HMI_INTERFACE_RC));
+  application_manager::SharedPtr<commands::ButtonPressRequest> command =
```

Comment:
@ValeriiMalkov use auto

Reply:
@LuxoftAKutsan fixed at the commit 051264163471fb3bb21b58ace83bbb7f9cf89b84